experimental/generation/generator/packages/library/templates/standard/form-BeginDialog.dialog.lg

```diff
@@ -13,18 +13,61 @@
             "$kind": "Microsoft.SendActivity",
             "activity": "\${welcome()}"           
         },
-        {
-            "$kind": "Microsoft.IfCondition",
-            "condition": "!turn.activityProcessed",
-            "actions": [
-                {
-                    "$kind": "Microsoft.EmitEvent",
-                    "eventName": "activityReceived",
-                    "eventValue": "=turn.activity",
-                    "bubbleEvent": false
-                }
-            ]
-        }
+        ${setActions()}
     ]
 }
 ```
+
+# setActions
+- IF: ${!empty(where(schema.properties, property, property.value.$dynamiclist == "true"))}
+-```{
+        "$kind": "Microsoft.IfCondition",
+        "condition": "!turn.activityProcessed",
+        "actions": [
+            {
+                "$kind": "Microsoft.EmitEvent",
+                "eventName": "activityReceived",
+                "eventValue": "=turn.activity",
+                "bubbleEvent": false
+            }
+        ]
+    },
+    ${setPropertiesForDynamiclists()}```
+- ELSE: 
+-``` {
+        "$kind": "Microsoft.IfCondition",
+        "condition": "!turn.activityProcessed",
+        "actions": [
+            {
+                "$kind": "Microsoft.EmitEvent",
+                "eventName": "activityReceived",
+                "eventValue": "=turn.activity",
+                "bubbleEvent": false
+            }
+        ]
+    }```
+
+# setPropertiesForDynamiclists()
+- ${join(foreach(where(schema.properties, property, property.value.$dynamiclist == "true"), result, `${setPropertyAction(result)}`), `,${EOL()}`)}
+
+# setPropertyAction(property)
+-```
+{
+    "$kind": "Microsoft.SetProperty",
+    "property": "dialog.${property.key}Value",
+    "value":  {
+    "entity": "${property.key}Value",
+    "list": [
+        ${listElements(property)}
+    ]
+    }
+}
+```
+
+# listElements(property)
+- IF: ${getProperty(property.value.$examples[""], concat(property.key, 'Value'))}
+    - ${join(foreach(indicesAndValues(getProperty(property.value.$examples[""], concat(property.key, 'Value'))), enum, `{"canonicalForm": "${enum.index}", "synonyms": [${synonymsForDynamicList(enum.value)}] }`), `, ${EOL()}`)}
+- ELSEIF: ${getProperty(property.value.$examples[locale], concat(property.key, 'Value'))}
+    - ${join(foreach(indicesAndValues(getProperty(property.value.$examples[locale], concat(property.key, 'Value'))), enum, `{"canonicalForm": "${enum.index}", "synonyms": [${synonymsForDynamicList(enum.value)}] }`), `, ${EOL()}`)}
```

Comment:
Use propertyList(property) to get the examples--it has the proper fallback logic.